From a423d6cba6c5960dc55b6e662ce58527abdccac1 Mon Sep 17 00:00:00 2001
From: Jian Cai <jiancai@google.com>
Date: Tue, 30 Mar 2021 16:02:49 -0700
Subject: [PATCH 11/12] blk-mq: fix alignment mismatch.

This fixes the mismatch of alignments between csd and its use as an
argument to smp_call_function_single_async, which causes build failure
when -Walign-mismatch in Clang is used.

Link:
http://crrev.com/c/1193732

Suggested-by: Guenter Roeck <linux@roeck-us.net>
Signed-off-by: Jian Cai <jiancai@google.com>
Link: https://lore.kernel.org/r/20210330230249.709221-1-jiancai@google.com
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 include/linux/blkdev.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index 542471b76f41..33c825c3f1bc 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -233,7 +233,7 @@ struct request {
 	unsigned long deadline;
 
 	union {
-		struct __call_single_data csd;
+		call_single_data_t csd;
 		u64 fifo_time;
 	};
 
-- 
2.31.0

